packages/commons/src/conditionals/conditionals.ts

```diff
@@ -21,11 +21,30 @@ export type JSONSchema = {
   readonly __nominal__type: 'JSONSchema'
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function removeIds(obj: any): any {
+  if (Array.isArray(obj)) {
+    return obj.map(removeIds)
+  } else if (obj !== null && typeof obj === 'object') {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const newObj: any = {}
+    for (const key of Object.keys(obj)) {
+      if (key === '$id') {
+        continue // skip $id
+      }
+      newObj[key] = removeIds(obj[key])
+    }
+    return newObj
+  }
+  return obj
+}
```

Comment:
Below is a refactored version of the function, with early returns and no `continue`. Wonder if we could get rid of the `any`s here?

```suggestion
// eslint-disable-next-line @typescript-eslint/no-explicit-any
function removeIds(obj: any): any {
  if (obj === null || typeof obj !== 'object') {
    return obj
  }

  if (Array.isArray(obj)) {
    return obj.map(removeIds)
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const newObj: any = {}
  for (const key of Object.keys(obj)) {
    if (key !== '$id') {
      newObj[key] = removeIds(obj[key])
    }
  }
  return newObj
}
```

I did not test if this works, so please do make sure :)

Reply:
Would be nice to have a comment for the function, telling what it does exactly and why its necessary.

Reply:
Thanks for your refactor suggestion, I have adjusted the changes, please have a look.